src/targets/gpu/include/migraphx/gpu/context.hpp

```diff
@@ -168,11 +169,40 @@ struct context
         return hip_event_ptr{event};
     }
 
+    value to_value() const
+    {
+        value result;
+        result["events"]  = events.size();
+        result["streams"] = current_device->nstreams();
+
+        return result;
+    }
+
+    void from_value(const value& v)
+    {
+        assert(v.contains("events"));
+        auto v_events = v.at("events");
+        assert(v_events.is_uint64());
+        std::size_t n_events = v_events.get_uint64();
+        this->create_events(n_events - 1);
+
+        assert(v.contains("streams"));
```

Comment:
There is no point in asserting this since `.at` will throw an exception if the key is missing.